src/diagnostics/wizardSteps/PromptSlotSelection.ts

```diff
@@ -49,22 +51,24 @@ export class PromptSlotSelection extends WizardStep {
             }
             throw e;
         }
-
-        this._wizard.selectedDeploymentSlot = pickedItem.data;
+        this._wizard.selectedDeploymentSlotTreeItem = pickedItem.data;
         this._wizard.writeline(`The deployment slot you selected is: ${this._wizard.selectedDeploymentSlot.name}`);
     }
+
     /**
      * Returns all the deployment slots and the production slot.
      */
-    private async getDeploymentSlots(): Promise<Site[]> {
-        const client = this._wizard.websiteManagementClient;
-        const allDeploymentSlots = await callWithTimeout(
-            () => {
-                return client.webApps.listByResourceGroup(this.site.resourceGroup, { includeSlots: true });
-            },
-            DEFAULT_TIMEOUT);
-        return allDeploymentSlots.filter((slot) => {
-            return slot.repositorySiteName === this.site.name;
-        });
+    private async getDeploymentSlotsTreeItems(): Promise<SiteTreeItem[]> {
+        const appServiceTreeItem = this._wizard.uiTreeItem.treeItem;
+        const result = await appServiceTreeItem.loadMoreChildren(<IAzureParentNode<SiteTreeItem>>this._wizard.uiTreeItem);
+
+        let deploymentSlotsCategoryNode: DeploymentSlotsTreeItem;
+        if (result && result.length > 0) {
+            deploymentSlotsCategoryNode = <DeploymentSlotsTreeItem>result[0];
```

Comment:
You know from the structure of the tree nodes that there will always be only a single category node at this level, and no other children?

Reply:
Not exactly, there are usually 3 categories. The first one is the deployment slots category, the other two are WebWorkers and AppSettings category.

I can do an `instanceof` style test to loop the three(or any).

Reply:
Yeah I wouldn't rely on it being the first node. We actually just added a fix recently to sort nodes in the tree alphabetically which would probably break you